table/options.go

```diff
@@ -336,6 +336,16 @@ func (m Model) WithFilterInputValue(value string) Model {
 	return m
 }
 
+// WithFilterFunc adds a filter function to the model. If the function returns a
+// truthy value, the row will be included in the filtered results. If the function
+// is nil, all rows will be included. The filter input is passed as the second
+// argument to the function.
+func (m Model) WithFilterFunc(filterFunc func(Row, string) bool) Model {
```

Comment:
Would prefer the args be named rather than only in the comment to explain them.  `row Row, filterInput string` would be fine.

Also wondering if we can make the name a little clearer for whether we need to return true or false, in hindsight.  Maybe as a nudge via intellisense:

```golang
func (m Model) WithFilterFunc(shouldInclude func(row Row, filterInput string) bool) Model {
```